src/main/java/spotify/factories/AbstractPlayableObjectDeserializer.java

```diff
@@ -7,13 +7,46 @@
 
 import java.lang.reflect.Type;
 
+/**
+ * This is a AbstractPlayableObjectDeserializer class to implement JsonDeserializer.
+ *
+ * @see spotify.models.generic.AbstractPlayableObject
+ * @author Jiankai Zheng (jk.zheng@hotmail.com)
+ * @since // check with Jiankai
+ */
+
 class AbstractPlayableObjectDeserializer implements JsonDeserializer<AbstractPlayableObject> {
+
+    /**
+     * Custom deserialize method to deserialize json, type and jsonDeserializationContext
+     * @param json represents a json element
+     * @param type represents various types
+     * @param jsonDeserializationContext
+     * @return childClass
+     * @throws JsonParseException (RuntimeException) if the incoming Json is bad/malicious/ serious issues that occurs during parsing of a Json string.
+     */
+
     @Override
     public AbstractPlayableObject deserialize(JsonElement json, Type type, JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
+        /**
```

Comment:
Writing javadoc for every line of code is way too extreme, this is not needed. Same goes for the javadoc under this line of code

Reply:
Thanks for your thorough feedback! I really appreciate it.